### PR TITLE
[W5.7][W16-1]Javen Seow Yu Bin

### PR DESCRIPTION
**Pull request by javenseow**
Added contain command

### CODE REVIEW

**Comment by kengwoon** [comment]
Can add more test cases, but other than that, all is good.

**Comment by GilgameshTC** [comment]
It would be better if the changes were done incrementally, but great job none-the-less!

**Comment by ericyjw** [comment]
Good job! However, you could have added comments before the methods!

File: src/seedu/addressbook/commands/ContainCommand.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public static final String MESSAGE_DUPLICATE = "This person is already in the address book";
+    public static final String MESSAGE_UNIQUE = "This person is not in the address book";
+
+    public ContainCommand(String name) throws IllegalValueException {
+        this.name = new Name(name);
+    }
+
+    @Override
+    public CommandResult execute() {
+        boolean personFound = checkDuplicate(name);
```
Comment (kengwoon) [inline src/seedu/addressbook/commands/ContainCommand.java]: can name variable as isPersonFound

**Review by kengwoon — APPROVED** [review]

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -97,12 +87,25 @@ public Command parseCommand(String userInput) {
...
+            return prepareContain(arguments);
+
         case HelpCommand.COMMAND_WORD: // Fallthrough
+
         default:
             return new HelpCommand();
         }
     }
 
+    private Command prepareContain(String args) {
```
Comment (bqnguyen94) [inline src/seedu/addressbook/parser/Parser.java]: Would be nice if this function has explanatory comments as well, see the `prepareAdd` function below :)

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (bqnguyen94) [inline src/seedu/addressbook/parser/Parser.java]: These style changes are not really relevant to the purpose of this PR which is adding the `contain` command.

File: test/expected.txt [hunk 4]
```diff
@@ -29,6 +29,7 @@
 || Example: help
 || exit: Exits the program.
 || Example: exit
+|| contain: Searches the address book with given name
```
Comment (bqnguyen94) [inline test/expected.txt:32]: How about an usage example as well? :)

File: test/java/seedu/addressbook/commands/ContainCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    @Test
+    public void execute_validAddressBook_personContained() throws IllegalValueException{
+        assertContainSuccessful("John Doe", addressbook, true);
+    }
+
+    @Test
+    public void execute_validAddressBook_personNotContained() throws IllegalValueException {
+        assertContainSuccessful("Lili Pan", addressbook, false);
+    }
+    private void assertContainSuccessful(String name, AddressBook addressbook, boolean contained) throws IllegalValueException {
```
Comment (bqnguyen94) [inline test/java/seedu/addressbook/commands/ContainCommandTest.java]: Nitpick: leave a line between functions to separate them.

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 6]
```diff
@@ -12,17 +12,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (bqnguyen94) [inline test/java/seedu/addressbook/parser/ParserTest.java]: These changes are also not really relevant?

**Review by bqnguyen94 — CHANGES_REQUESTED** [review]
Almost there 👍

**Review by bqnguyen94 — CHANGES_REQUESTED** [review]
Pls update the user guide as well

File: test/expected.txt (hunk 4 above)
Comment (javenseow) [reply test/expected.txt:32]: Added an example!

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 7]
```diff
@@ -23,6 +24,7 @@
 import seedu.addressbook.commands.ListCommand;
 import seedu.addressbook.commands.ViewAllCommand;
 import seedu.addressbook.commands.ViewCommand;
+
```
Comment (bqnguyen94) [inline test/java/seedu/addressbook/parser/ParserTest.java:27]: This is not needed. Normally we only separate imports that are from different parties.

**Review by bqnguyen94 — APPROVED** [review]
Good job 👍 you can close the PR now.

**Pull request closed without merging**